### PR TITLE
Allow entire PodSpec to be defined in Run Collector

### PR DESCRIPTION
**Pull request by diamonwiggins**

### CODE REVIEW

**Comment by diamonwiggins** [comment]
Should we still support the creation of an image pull secret with this method?

**Comment by dexhorthy** [comment]
Awesome! 🚀

File: pkg/collect/run.go [hunk 1]
```diff
@@ -140,38 +140,55 @@ func runPod(ctx context.Context, client *kubernetes.Clientset, runCollector *tro
...
 			},
-		},
-	}
+		}
 
-	if runCollector.ImagePullSecret != nil && runCollector.ImagePullSecret.Data != nil {
-		secretName, err := createSecret(ctx, client, pod.Namespace, runCollector.ImagePullSecret)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to create secret")
+		if runCollector.ImagePullSecret != nil && runCollector.ImagePullSecret.Data != nil {
```
Comment (emosbaugh) [inline pkg/collect/run.go:183]: is this useful when there is a podspec?

File: pkg/apis/troubleshoot/v1beta2/collector_shared.go [hunk 2]
```diff
@@ -75,6 +76,7 @@ type Run struct {
 	ImagePullPolicy    string            `json:"imagePullPolicy,omitempty" yaml:"imagePullPolicy,omitempty"`
 	ImagePullSecret    *ImagePullSecrets `json:"imagePullSecret,omitempty" yaml:"imagePullSecret,omitempty"`
 	ServiceAccountName string            `json:"serviceAccountName,omitempty" yaml:"serviceAccountName,omitempty"`
+	PodSpec            corev1.PodSpec    `json:"podSpec,omitempty" yaml:"podSpec,omitempty"`
```
Comment (emosbaugh) [inline pkg/apis/troubleshoot/v1beta2/collector_shared.go:79]: The only thing i question is whether or not we should create an entirely new collector like RunPod... Did you consider that? Do you think reusing this is better?

Comment (diamonwiggins) [reply pkg/apis/troubleshoot/v1beta2/collector_shared.go:79]: I think reusing is worth it if it isn't confusing for users to discern which properties are used based on when you use `podSpec` or not. I do question that. 🤔

File: pkg/collect/run.go (hunk 1 above)
Comment (diamonwiggins) [reply pkg/collect/run.go:183]: when `podSpec` isn't used you'd still be able to use this feature, but planned to make it clear in the docs that with `podSpec` you'd always have to specify an existing secret name in the spec

**Pull request closed without merging**